### ISSUE TITLE
String literal constant folding

### ISSUE THREAD

**Issue by Dan12** [polyglot-compiler/JLang#34]
It appears that string literals don't have constant folding performed on them, although it might be a different issue. The following code produces two different results
```java
System.out.println("hello" == "hello"); // prints true
System.out.println("hello" == "hel"+"lo"); // prints false
```
however, it would seem that with constant folding they should produce the same result.

**Comment by Dan12** [comment]
This issue has been tracked down to Polyglot's ConstantFolder.java visitor explicitly ignoring string concatenation.
We could implement a full java constant folder for Polyglot and invoke it as an optimization pass.

**Comment by dz333** [comment]
This has been fixed in merge pull request #36
